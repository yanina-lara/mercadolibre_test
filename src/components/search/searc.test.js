import React from "react";
import { render, screen, fireEvent } from '@testing-library/react';
import SearchInput from "./searchInput";
import { MemoryRouter } from 'react-router-dom';

describe('Test de SearchInput', () => {
  it('Display component', () => {
    const {baseElement: searchInputComponent} = render(<SearchInput/>, {wrapper: MemoryRouter});
    expect(searchInputComponent).toMatchSnapshot();
    fireEvent.submit(screen.getByTestId('search-form'));
    fireEvent.change(screen.getByTestId('search-input'), {target:{value:'test'}});
  });
  it('Display component 2', () => {
    const {baseElement: searchInputComponent} = render(<SearchInput searchValueDefault="test"/>, {wrapper: MemoryRouter});
    expect(searchInputComponent).toMatchSnapshot();
    fireEvent.submit(screen.getByTestId('search-form'));
  });
});
