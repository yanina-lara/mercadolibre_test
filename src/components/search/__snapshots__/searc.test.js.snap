// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de SearchInput Display component 1`] = `
<body>
  <div>
    <div
      class="search"
    >
      <div
        class="container"
      >
        <form
          data-testid="search-form"
        >
          <div
            class="input_box"
          >
            <a
              href="/"
            >
              <img
                class="logo"
                src="logo.png"
              />
            </a>
            <input
              class=""
              data-testid="search-input"
              id="inlineFormInputGroupUsername"
              placeholder="Nunca dejes de buscar"
              type="text"
              value=""
            />
            <button
              type="submit"
            >
              <img
                height="15px"
                src="search_icon.png"
                width="15px"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</body>
`;

exports[`Test de SearchInput Display component 2 1`] = `
<body>
  <div>
    <div
      class="search"
    >
      <div
        class="container"
      >
        <form
          data-testid="search-form"
        >
          <div
            class="input_box"
          >
            <a
              href="/"
            >
              <img
                class="logo"
                src="logo.png"
              />
            </a>
            <input
              class=""
              data-testid="search-input"
              id="inlineFormInputGroupUsername"
              placeholder="Nunca dejes de buscar"
              type="text"
              value="test"
            />
            <button
              type="submit"
            >
              <img
                height="15px"
                src="search_icon.png"
                width="15px"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</body>
`;
